### PR TITLE
Treat comma as a valid character in decimal number strings, closes #6028

### LINKED ISSUE #6028: Interpret comma as decimal point in editor
**Operating system or device - Godot version:**
Linux (Manjaro) - v2.2 alpha (git custom build)

**Issue description** (what happened, and what was expected):
If you use a German keyboard layout (and maybe other layouts), the "decimal point" is a comma. If you insert numbers into the editor's fields (e.g. the translation) it would be nice if this comma would be automatically replaced by a decimal point. Blender does the same and it this little fix would enhance the workflow.

**Steps to reproduce:**

**Link to minimal example project** (optional but very welcome):

### PR DESCRIPTION
**Pull request by ghost**
Works in the Editor and in scenes.

Test scene:  [comma.zip](https://github.com/godotengine/godot/files/411568/comma.zip)

### CODE REVIEW

File: core/ustring.cpp [hunk 1]
```diff
@@ -1594,7 +1594,8 @@ int String::to_int() const {
 	if (length()==0)
 		return 0;
 
-	int to=(find(".")>=0) ? find(".") : length() ;
+	int dot = MAX(find("."),find(",");
```
Comment (neikeq) [inline core/ustring.cpp]: Missing `)`.

File: core/ustring.cpp [hunk 2]
```diff
@@ -1622,7 +1623,8 @@ int64_t String::to_int64() const {
...
```
Comment (neikeq) [inline core/ustring.cpp]: Same here.

**Comment by rollenrolm** [comment]
Updated the patch to address the issues and rebased on current master.

**Comment by reduz** [comment]
I'm not sure about this one, do other softwares also understand it as such?

**Comment by vnen** [comment]
@reduz according to #6028, Blender does such thing, but this is related to i18n, so it should be editor only.

I think this should be a setting in editor to select the decimal separator and proper support different locales, without changing how GDScript works.

**Comment by reduz** [comment]
Yeah I guess we should do some sort of str.replace(",",".") in the few GUI
places where numeric text is entered (SpinBox and Tree?) instead

**Review by akien-mga — CHANGES_REQUESTED** [review]
As commented by @reduz and @vnen, the proper fix would be to convert commas to dots in the input fields of the editor, and not to add support for commas in ustring.

**Comment by akien-mga** [comment]
Closing as per the above, don't hesitate to make a new PR implementing the conversion in the editor input fields instead of in ustring.

**Pull request closed without merging**